### PR TITLE
Add save and display of per-protein read lengths in HTML output of proteins-to-pathogens.py if --showReadLengths is given

### LINKED ISSUE #537: Make it possible to display read lengths in the HTML output of proteins-to-pathogens.py
Sofia wants this. Should be off by default (as many reads can match a protein and it doesn't seem generally useful to dump out hundreds or thousands of read lengths).

### PR DESCRIPTION
**Pull request by terrycojones**
Fixes #537

### CODE REVIEW

File: dark/proteins.py [hunk 1]
```diff
@@ -302,11 +305,13 @@ def addFile(self, filename, fp):
                     'Protein %r already seen for pathogen %r sample %r.' %
                     (proteinName, pathogenName, sampleName))
 
+            readsFilename = join(outDir, '%d.%s' % (index, self._format))
```
Comment (bamueh) [inline dark/proteins.py]: Why are you defining another variable? What's wrong with the previous version?

File: dark/proteins.py [hunk 2]
```diff
@@ -486,11 +497,18 @@ def toHTML(self, pathogenPanelFilename=None, minProteinFraction=0.0,
             '<li>Read count.</li>',
             '<li>HSP count (a read can match a protein more than once).</li>',
             '<li>Protein length (in AAs).</li>',
-            '<li>Index (just ignore this).</li>',
+            '<li>Index (just ignore this).</li>'
```
Comment (bamueh) [inline dark/proteins.py]: There should be a , after this line.

File: dark/proteins.py (hunk 1 above)
Comment (terrycojones) [inline dark/proteins.py]: Because I re-use the filename in the subsequent code (if read lengths are being saved).

File: dark/proteins.py (hunk 2 above)
Comment (terrycojones) [inline dark/proteins.py]: Fixed, thanks.

File: dark/proteins.py (hunk 1 above)
Comment (bamueh) [inline dark/proteins.py]: Ah, I see.

**Pull request merged**